File: base/src/main/java/com/github2136/base/IBaseMVPView.java
package com.github2136.base;

import android.support.annotation.StringRes;

/**
 *
 */
public interface IBaseMVPView {
    // 显示进度框
    void showProgressDialog();

    // 显示进度框
    void showProgressDialog(@StringRes int resId);

    // 显示进度框
    void showProgressDialog(String msg);

    // 关闭进度框
    void dismissDialog();
}

File: base/src/main/java/com/github2136/base/IBaseMVPView.kt
package com.github2136.base

import androidx.annotation.StringRes

/**
 * Created by yb on 2018/11/1.
 */
interface IBaseMVPView {
    // 显示进度框
    fun showProgressDialog()

    // 显示进度框
    fun showProgressDialog(@StringRes resId: Int)

    // 显示进度框
    fun showProgressDialog(msg: String)

    // 关闭进度框
    fun dismissDialog()
}